src/units/filter.cpp

```diff
@@ -425,18 +425,28 @@ void unit_filter_compound::fill(const vconfig& cfg)
 						}
 					}
 
-					const auto adjacent = get_adjacent_tiles(args.loc);
-					for(unsigned i = 0; i < adjacent.size(); ++i) {
-						const unit_map::const_iterator it = units.find(adjacent[i]);
-						if (it == units.end() || it->incapacitated())
+					for(const unit& unit : units) {
+						if(!unit.has_ability_distant() || unit.incapacitated() || &unit == args.u.shared_from_this().get()) {
 							continue;
-						if (&*it == (args.u.shared_from_this()).get())
+						}
+						const map_location& from_loc = unit.get_location();
+						unsigned int distance = distance_between(from_loc, args.loc);
```

Comment:
Similar comment here - practically speaking it doesn't matter, but the return type of `distance_between()` is `std::size_t`, not `unsigned`.